core/src/main/java/com/datastax/oss/driver/internal/core/session/SchemaListenerNotifier.java

```diff
@@ -33,6 +34,11 @@
   private final SchemaChangeListener listener;
   private final EventExecutor adminExecutor;
 
+  // It is technically possible that a schema change could happen in the middle of session
+  // initialization. Don't forward events in this case, it would likely do more harm than good if a
+  // listener implementation doesn't expect it.
+  private boolean sessionReady;
```

Comment:
Unlike node events, I think it's highly unlikely that anyone would be interested in schema changes that happened before the session was ready. So I changed the driver internals to ignore them, instead of introducing a replay mechanism.